src/Config.php

```diff
@@ -6,59 +6,56 @@
 use Symfony\Component\Yaml\Yaml;
 use Terminus\Exceptions\TerminusException;
 
-class Config
+class Config extends \Robo\Config
 {
     /**
-     * @var string[]
+     * @var boolean
      */
-    private static $config = [];
+    private $configured = false;
+
     /**
      * @var string
      */
-    private static $config_path = '/config/constants.yml';
+    private $config_path = '/config/constants.yml';
     /**
      * @var string
      */
-    private static $constant_prefix = 'TERMINUS_';
+    private $constant_prefix = 'TERMINUS_';
 
     /**
      * Constructor for Config
-     *
-     * @param array $options Options with which to configure this object
      */
-    public function __construct(array $options = [])
+    public function __construct()
     {
-        self::$config = $options;
     }
 
     /**
-     * Returns a configuration setting
-     *
-     * @param string $key The key of the config setting to return
-     * @return string self::$config[$property]
-     * @throws TerminusException
+     * @inheritdoc
      */
-    public static function get($key)
+    public function getGlobalOptionDefaultValues()
     {
-        $config = self::getAll();
-        try {
-            return $config[$key];
-        } catch (\Exception $e) {
-            throw new TerminusException('No configuration setting for {key} found.', compact('key'));
-        }
+        // We deliberately do not want to return any of the values from
+        // the parent function here.  Any global CLI options that Terminus
+        // adds should be defined here in an associative array that
+        // returns the key => default value of the global options.
+        return [];
     }
 
     /**
-     * Returns all configuration settings
-     *
-     * @return string[] self::$config
+     * @inheritdoc
+     * @throws TerminusException
      */
-    public static function getAll()
+    public function get($key, $defaultOverride = null)
     {
-        if (empty(self::$config)) {
-            self::configure();
+        if (!$this->configured) {
```

Comment:
I prefer to write constructors that are very simple, e.g. no file access, and lazy-construct everything else. If you don't mind constructors that can fail, then it would work just fine to load up the config values in the constructors.

Perhaps the best thing to do would be to make a config loader class, use it to get all of the config values (from disk et. al.), and then inject the config values. That wouldn't be lazy-loaded, but that's not needed here, and it would keep io out of the constructor.